src/core/net_db.cc

```diff
@@ -154,34 +154,30 @@ void NetDb::Run() {
       }
       // publishes router info to a floodfill at Nth interval
       if (ts - last_publish >=
-	  static_cast<std::uint16_t>(NetDbDuration::PublishRouterInfoInterval)) {
+	  static_cast<std::uint16_t>(NetDbInterval::PublishRouterInfo)) {
         Publish();
         last_publish = ts;
       }
       // builds exploratory tunnels at Nth interval to find more peers
-      if (ts - last_exploratory >=
-          static_cast<std::uint16_t>(NetDbDuration::ExploreInterval)) {
-        auto known_routers = m_RouterInfos.size();
+      if (ts - last_exploratory >= static_cast<std::uint16_t>(NetDbInterval::Exploratory)) {
+        auto known_routers = GetNumRouters();
         std::uint16_t num_routers = 0;
         // evaluates if a router has a sufficient number of known routers
         // to use for building tunnels, if less than Nth routers are known,
         // then more exploratory tunnels will be created to find more routers
         // for tunnel building
-        if (known_routers > 0 && known_routers <
-            static_cast<std::uint16_t>(NetDbPrefs::MinimumKnownRouters)) {
-          num_routers =
-            static_cast<std::uint16_t>(NetDbPrefs::MaxExploratoryTunnels);
-        } else if ((known_routers >
-            static_cast<std::uint16_t>(NetDbPrefs::MinimumKnownRouters) &&
-            known_routers <
-            static_cast<std::uint16_t>(NetDbPrefs::FavouredNumKnownRouters)) ||
-            ts - last_exploratory >=
-            static_cast<std::uint16_t>(NetDbDuration::DelayedExploreInterval)) {
-          num_routers = static_cast<std::uint16_t>(NetDbPrefs::MinExploratoryTunnels);
+        if (known_routers > 0
+            && known_routers < static_cast<std::uint16_t>(NetDbSize::MinKnownRouters)) {
+          num_routers = static_cast<std::uint16_t>(NetDbSize::MaxExploratoryTunnels);
+        } else if (!known_routers
```

Comment:
This was at the end of a long day, I took the meaning out of context. I'll revert. Fortunately, I've ensured that we always have enough routers to begin (in reseed).